### PR TITLE
[compiler] Support default imports for autodep config

### PR DESCRIPTION
**Pull request by jbrown215**
## Summary

Allows us to add deps for things like `import useWrapperEffect from 'useWrapperEffect'`
---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/facebook/react/pull/31657).
* __->__ #31657
* #31652

### CODE REVIEW

File: compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/infer-effect-dependencies.expect.md [hunk 1]
```diff
@@ -125,6 +131,17 @@ function Component(t0) {
...
+  if ($[12] !== foo) {
+    t6 = () => {
+      console.log(foo);
+    };
+    $[12] = foo;
+    $[13] = t6;
+  } else {
+    t6 = $[13];
+  }
+  useEffectWrapper(t6, [foo]);
```
Comment (mofeiZ) [inline compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/infer-effect-dependencies.expect.md:144]: Nice 👍

**Review by mofeiZ — APPROVED** [review]
Makes sense!

File: compiler/packages/babel-plugin-react-compiler/src/Inference/InferEffectDependencies.ts [hunk 2]
```diff
@@ -97,6 +97,17 @@ export function inferEffectDependencies(fn: HIRFunction): void {
             autodepFnLoads.set(lvalue.identifier.id, numRequiredArgs);
           }
         }
+      } else if (
+        value.kind === 'LoadGlobal' &&
+        value.binding.kind === 'ImportDefault'
+      ) {
+        const moduleTargets = autodepFnConfigs.get(value.binding.module);
+        if (moduleTargets != null) {
+          const numRequiredArgs = moduleTargets.get('default');
```
Comment (mofeiZ) [inline compiler/packages/babel-plugin-react-compiler/src/Inference/InferEffectDependencies.ts:106]: Does it make sense to make `default` a constant in `Environment` for grep-ability?

Comment (jbrown215) [reply compiler/packages/babel-plugin-react-compiler/src/Inference/InferEffectDependencies.ts:106]: yea, sounds reasonable to me! I like grep-ability + find-refs-ability

**Pull request merged**